src/Compilers/CSharp/Portable/Binder/Binder_TupleOperators.cs

```diff
@@ -0,0 +1,284 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class Binder
+    {
+        /// <summary>
+        /// If the left and right are tuples of matching cardinality, we'll try to bind the operator element-wise.
+        /// When that succeeds, the element-wise conversions are collected and their result type is applied as tuple conversions.
+        /// The element-wise binary operators are collected and stored as a tree for lowering.
+        /// </summary>
+        private BoundTupleBinaryOperator BindTupleBinaryOperator(BinaryExpressionSyntax node, BinaryOperatorKind kind,
+            BoundExpression left, BoundExpression right, DiagnosticBag diagnostics)
+        {
+            TupleBinaryOperatorInfo operators = BindTupleBinaryOperatorInfo(node, kind, left, right, diagnostics);
+
+            BoundExpression convertedLeft = GenerateConversionForAssignment(operators.LeftConvertedType, left, diagnostics);
+            BoundExpression convertedRight = GenerateConversionForAssignment(operators.RightConvertedType, right, diagnostics);
+
+            TypeSymbol resultType = GetSpecialType(SpecialType.System_Boolean, diagnostics, node);
+            return new BoundTupleBinaryOperator(node, convertedLeft, convertedRight, kind, operators, resultType);
+        }
+
+        private TupleBinaryOperatorInfo BindTupleBinaryOperatorInfo(BinaryExpressionSyntax node, BinaryOperatorKind kind,
+            BoundExpression left, BoundExpression right, DiagnosticBag diagnostics)
+        {
+            int leftCardinality = GetTupleCardinality(left).GetValueOrDefault();
+            int rightCardinality = GetTupleCardinality(right).GetValueOrDefault();
+
+            if (leftCardinality > 1 && rightCardinality > 1)
```

Comment:
>if (leftCardinality > 1 && rightCardinality > 1) [](start = 12, length = 48)

It feels like the body of this ```if``` should be extracted into its own method returning TupleBinaryOperatorInfo.Multiple, which should be called directly by BindTupleBinaryOperator instead of calling containing method